.github/workflows/performance-regression.yml

```diff
@@ -0,0 +1,146 @@
+name: Performance Regression Detection
+on:
+  pull_request:
+    branches: [ master ]
+jobs:
+  detect-regression:
+    name: Detect Regression
+    runs-on: ubuntu-latest
+    strategy:
+     fail-fast: true
+    env:
+      CC: 'clang'
+      CXX: 'clang++'
+    steps:
+      - name: Install Dependencies
+        run: |
+          apt-get update
+          apt-get install git cmake clang-14 python3-pip -y
+          [[ -e /usr/bin/cmake ]] || ln -s `which cmake3` /usr/bin/cmake
+
+      - name: Get Data Generator
+        uses: actions/checkout@v2
```

Comment:
Could we pin to a specific commit instead of commit hash tag? Tags can be updated and re-pointed, so the code referenced by a tag can change. However the commit SHA represents the exact snapshot of the code at a point in time. 

The latest version of actions/checkout is recommended. `actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1`
Same suggestion to other action/checkout call.